### ISSUE TITLE
Selection of Weights 

### ISSUE THREAD

**Issue by javierpastorfernandez** [stardist/stardist#128]
Good morning,

I  realized that StarDist usually saves two versions for the weights after the training (Weights_best, weights_last /weights_now) and uses weight_best by default. 

I am wondering what is the internal process that StarDist /CsbDeep Follows to create weights_best. Is it a training metric of the VALIDATION DATA?  

(I have revised the StarDist source code, as well as CSBDeep's, but I cannot find it! Thank you :)

![image](https://user-images.githubusercontent.com/60293572/111900163-3ad4fe80-8a31-11eb-98f0-410e11e7a4d5.png)

**Comment by uschmidt83** [comment]
Hi,

> I am wondering what is the internal process that StarDist /CsbDeep Follows to create weights_best. Is it a training metric of the VALIDATION DATA?

the files `weights_best.h5` and `weights_now.h5` (names defined [here](https://github.com/CSBDeep/CSBDeep/blob/fc4479f74d04eebcf871e2957e0ed9dff65a03a4/csbdeep/models/config.py#L54-L56)) are saved [via Keras callbacks](https://github.com/CSBDeep/CSBDeep/blob/fc4479f74d04eebcf871e2957e0ed9dff65a03a4/csbdeep/models/base_model.py#L187-L196) during training. By default (and what we use), the [`ModelCheckpoint`](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/ModelCheckpoint) callback saves the "best" weights according to the validation loss. The weights `weights_last.h5` at the end of training are saved [here](https://github.com/CSBDeep/CSBDeep/blob/master/csbdeep/models/base_model.py#L199-L211).
